backend/app/services/execution/request_builder.py

```
@@ -1275,6 +1293,186 @@ def _build_mcp_servers(self, bot: Kind, team: Kind) -> list[dict]:
+        Args:
+            server: Server config dict with 'url' and optional 'headers'
+
+        Returns:
+            True if server responded, False on connection failure
+        """
+        url = server.get("url", "")
+        if not url:
+            return False
+
+        try:
+            req = urllib.request.Request(url, method="HEAD")
```

Ruff (0.15.4)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)